### ISSUE TITLE
0.8.0-rc3 IllegalArgumentException using JerseyClient with a webtarget having a registered component

### ISSUE THREAD

**Issue by ESPNMichaelKidd** [dropwizard/dropwizard#916]
```java
Client client = new JerseyClientBuilder(environment).build("jerseyclient");
        
WebTarget wt = client.target("http://google.com");
wt.register(new LoggingFilter(java.util.logging.Logger.getLogger("loggingfilter"), false));
wt.request().get(String.class);

WebTarget wt2 = client.target("http://google.com");
wt2.request().get(String.class);
```

the 2nd get will fail with he following exception:
```
Exception in thread "main" java.lang.IllegalArgumentException: A metric named org.apache.http.conn.HttpClientConnectionManager.jerseyclient.available-connections already exists
	at com.codahale.metrics.MetricRegistry.register(MetricRegistry.java:91)
	at com.codahale.metrics.httpclient.InstrumentedHttpClientConnectionManager.<init>(InstrumentedHttpClientConnectionManager.java:57)
	at io.dropwizard.client.HttpClientBuilder.createConnectionManager(HttpClientBuilder.java:278)
	at io.dropwizard.client.HttpClientBuilder.build(HttpClientBuilder.java:173)
	at io.dropwizard.client.JerseyClientBuilder$1.getConnector(JerseyClientBuilder.java:317)
	at org.glassfish.jersey.client.ClientConfig$State.initRuntime(ClientConfig.java:414)
	at org.glassfish.jersey.client.ClientConfig$State.access$000(ClientConfig.java:87)
	at org.glassfish.jersey.client.ClientConfig$State$3.get(ClientConfig.java:119)
	at org.glassfish.jersey.client.ClientConfig$State$3.get(ClientConfig.java:116)
	at org.glassfish.jersey.internal.util.collection.Values$LazyValueImpl.get(Values.java:340)
	at org.glassfish.jersey.client.ClientConfig.getRuntime(ClientConfig.java:722)
	at org.glassfish.jersey.client.ClientRequest.getConfiguration(ClientRequest.java:284)
	at org.glassfish.jersey.client.JerseyInvocation.validateHttpMethodAndEntity(JerseyInvocation.java:126)
	at org.glassfish.jersey.client.JerseyInvocation.<init>(JerseyInvocation.java:98)
	at org.glassfish.jersey.client.JerseyInvocation.<init>(JerseyInvocation.java:91)
	at org.glassfish.jersey.client.JerseyInvocation$Builder.method(JerseyInvocation.java:408)
	at org.glassfish.jersey.client.JerseyInvocation$Builder.get(JerseyInvocation.java:308)
```
This does not fail in 0.8.0-rc2.
This fails in the same manner in rc4 & rc5

**Comment by ESPNMichaelKidd** [comment]
as best I can tell, I believe the issue is introduced here: https://github.com/dropwizard/dropwizard/commit/48fdb73ebb24aafd4ed18e968e01363ea6abada0#diff-c810a2744e3beb152ebc3f6052aa5509R317

as doing this more than one will result in the illegal argument exception
```
new DropwizardApacheConnector(new HttpClientBuilder(environment).build("name"),true);
```

**Comment by arteam** [comment]
@ESPNMichaelKidd, thanks for reporting this.

The problem with the code is that `LoggingFilter` is set on a web target, rather than on a client.
If the filter is set on the client, the code works fine:
````java
Client client = new JerseyClientBuilder(environment).build("test-jersey-client");
client.register(new LoggingFilter());

WebTarget wt = client.target("http://google.com");
String firstResponse = wt.request().get(String.class);
System.out.println(firstResponse);

WebTarget wt2 = client.target("http://google.com");
String secondResponse = wt2.request().get(String.class);
System.out.println(secondResponse);
````

Anyway, it's a legal configuration and we should handle cases like that.
I've made `ConnectorProvider` idempotent. After that `JerseyClient` should handle configurations like that (with filters on targets). 

Could you test this againt the latest snapshot?

**Comment by ESPNMichaelKidd** [comment]
@arteam fixed in latest snapshot.  Thanks!

**Comment by arteam** [comment]
Great. The fix will be avaiable in the upcoming 0.8.0.
